core/src/main/java/org/apache/accumulo/core/client/sample/AbstractHashSampler.java

```diff
@@ -57,12 +58,33 @@
   private int modulus;
 
   private static final Set<String> VALID_OPTIONS = Set.of("hasher", "modulus");
+  private static final Set<String> VALID_VALUES_HASHER = Set.of("murmur3_32", "md5", "sha1");
+
+  /**
+   * Subclasses with options should override this method to validate subclass options while also
+   * calling {@code super.validateOptions(config)} to validate base class options.
+   */
+  @Override
+  public void validateOptions(Map<String,String> config) {
+    for (Map.Entry<String,String> entry : config.entrySet()) {
+      checkArgument(VALID_OPTIONS.contains(entry.getKey()), "Unknown option: %s", entry.getKey());
+
+      if (entry.getKey().equals("hasher"))
+        checkArgument(VALID_VALUES_HASHER.contains(entry.getValue()),
+            "Unknown value for hasher: %s", entry.getValue());
+
+      if (entry.getKey().equals("modulus"))
+        checkArgument(Integer.parseInt(entry.getValue()) > 0,
+            "Improper Integer value for modulus: %s", entry.getValue());
+    }
+  }
 
   /**
    * Subclasses with options should override this method and return true if the option is valid for
    * the subclass or if {@code super.isValidOption(opt)} returns true.
+   * @deprecated since 2.1.0, replaced by {@link #validateOptions(Map)}
```

Comment:
```suggestion
   *
   * @deprecated since 2.1.0, replaced by {@link #validateOptions(Map)}
```

Reply:
That is what caused the other build error. I couldn't win with either.

Reply:
I guess it was something on my end to create extra white space. I had what you committed before and the build failed.

Reply:
When the formatter automatically adds a line, it leaves a space after the `*` to indent the section. That causes checkstyle to complain. However, if you delete that extra space, the formatter is fine with it and will not reformat it to add it again. My change added the line back without the extra space that caused it to fail previously. It's just an annoying quirk of the formatter.

Reply:
I actually think it might be useful to add the trailing space removal as a feature to https://github.com/revelc/formatter-maven-plugin since the Eclipse formatter library it uses doesn't have a feature to remove the trailing whitespace itself.

Reply:
> When the formatter automatically adds a line, it leaves a space after the * to indent the section. That causes checkstyle to complain. However, if you delete that extra space, the formatter is fine with it and will not reformat it to add it again. My change added the line back without the extra space that caused it to fail previously. It's just an annoying quirk of the formatted.

ah, that makes sense. Thanks for fixing it.